### PR TITLE
Feature/nav bar

### LINKED ISSUE #5: Navigation Bar
- Build the Navigation bar based on the figma design
- **New: ** Do the condensed/hamburger menu version of nav bar

### PR DESCRIPTION
**Pull request by leo3friedman**
## Tracking Info

Resolves #5

## Changes

- Added nav bar component and button component. 
- Added Roboto and Rubik fonts.
- Removed Vite starter code. 

## Testing

- Manual testing

## Confirmation of Change

TODO

### CODE REVIEW

**Review by kathyychenn — APPROVED** [review]

**Review by jaredsbt — APPROVED** [review]

**Review by Miyuki-L — CHANGES_REQUESTED** [review]
Overall Good work for the first task so far. Lets get these changes implemented first. 

Also lets add this line `// eslint-disable-next-line @typescript-eslint/no-non-null-assertion` above line 7 of main.tsx . Root element is never null.

File: package-lock.json
Comment (Miyuki-L) [inline package-lock.json:1]: This and package.json should be deleted. These should be in the frontend folder. 
WHen you ran npm install react-router-dom you did it in the root directory and so the project on my side is also not compiling.

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -1,39 +1,42 @@
...
 
-import reactLogo from "./assets/react.svg";
-
-import viteLogo from "/vite.svg";
-import "./App.css";
-
-function App() {
-  const [count, setCount] = useState(0);
+import NavBar from "./components/NavBar.js";
+import AboutUs from "./pages/AboutUs.tsx";
```
Comment (Miyuki-L) [inline frontend/src/App.tsx:4]: One last thing lets add the exports for the pages and nav like how Kathy and Jared did for the pages. [here](https://github.com/TritonSE/CCIDC-Application-Overhaul/blob/7-home-page/frontend/src/pages/index.ts) One index.ts file for each folder and update the import statements.

**Review by Miyuki-L — CHANGES_REQUESTED** [review]
Fantastic changes

**Comment by leo3friedman** [comment]
ccidc.org navbar does not have link addresses for the resources nav, search, or cart. In our nav, search and cart do nothing and resources links to the base ccidc.org website.

**Review by Miyuki-L — APPROVED** [review]

**Pull request merged**